### PR TITLE
Improve confluence tests

### PR DESCRIPTION
**Pull request by fizruk**
1. Speed up confluence tests by using a better order checks in the intersection.
2. Fix Rule 2 to be applicable inside applications and dispatches.
3. Remove `genericShrink` from attribute names and bytes. These should not shrink.
4. Change arbitrary names to be two-symbol (letter + number), e.g. `a3` or `m7`.
5. Show inputs that result in timeouts.

---

## PR-Codex overview
The focus of this PR is to refactor tests and update language features. 

### Detailed summary
- Refactored test structure for `eo-phi-normalizer`
- Added language features like `LambdaCase` and `TupleSections`
- Improved object generation logic for testing

> The following files were skipped due to too many changes: `eo-phi-normalizer/test/Language/EO/Rules/PhiPaperSpec.hs`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo-phi-normalizer/test/Language/EO/Rules/PhiPaperSpec.hs [hunk 1]
```diff
@@ -35,51 +40,59 @@ instance Arbitrary Attribute where
...
+        arbitraryAlphaLabelBinding =
+          resize (n `div` 2) $
+            AlphaBinding <$> (Label <$> arbitrary) <*> arbitrary
+        arbitraryAlphaLabelBindings = List.nubBy sameAttr <$> listOf arbitraryAlphaLabelBinding
     if n > 0
       then
         oneof
           [ Formation <$> arbitraryBindings
-          , liftA2 Application arbitraryObj arbitraryBindings
+          , liftA2 Application arbitraryObj arbitraryAlphaLabelBindings
```
[the remark this answers is not in the document]
Comment (fizruk) [reply eo-phi-normalizer/test/Language/EO/Rules/PhiPaperSpec.hs:95]: Application can only happen for attributes that can be in `EmptyBinding`s (or $\rho$, but that's only for internal use).

File: eo-phi-normalizer/test/eo/phi/rules/yegor.yaml [hunk 2]
```diff
@@ -90,7 +91,10 @@ rules:
       Φ.!a
     result: |
       ⟦ σ ↦ Φ, !B ⟧.!a
-    when: []
+    when:
+      - present_attrs:
+          attrs: ['ν']
+          bindings: ['!B']
```
[the remark this answers is not in the document]
Comment (fizruk) [reply eo-phi-normalizer/test/eo/phi/rules/yegor.yaml]: Another attempt to fix rule 4. As discussed elsewhere, we will now try to go with a different approach.

[the remark this answers is not in the document]
Comment (aabounegm) [reply eo-phi-normalizer/test/eo/phi/rules/yegor.yaml]: "now" in this PR or another one?

[the remark this answers is not in the document]
Comment (fizruk) [reply eo-phi-normalizer/test/eo/phi/rules/yegor.yaml]: This requires working out proper rules first. So we will implement that separately.

**Comment by 0pdd** [comment]
@fizruk the puzzle [#175](https://github.com/objectionary/normalizer/issues/175) is still not solved.

**Comment by 0pdd** [comment]
@fizruk the only puzzle [#175](https://github.com/objectionary/normalizer/issues/175) is solved here.

**Pull request merged**